### PR TITLE
Drtii 867 port filter

### PR DESCRIPTION
**Pull request by ngu04**
- adding destination port filter for IEI

### CODE REVIEW

File: src/main/scala/uk/gov/homeoffice/drt/api/ArrivalRoutes.scala [hunk 1]
```diff
@@ -25,10 +25,11 @@ object ArrivalRoutes {
           val xAuthRoles: List[String] = req.headers.get(CaseInsensitiveString("X-Auth-Roles")).map(_.value.split(",").toList).getOrElse(List.empty)
           val xAuthEmail: List[String] = req.headers.get(CaseInsensitiveString("X-Auth-Email")).map(_.value.split(",").toList).getOrElse(List.empty)
           val requiredPermissions: Boolean = xAuthRoles.exists(p => permissions.contains(p))
+          val portList = params.getOrElse("portList", Seq.empty[String]).toList.filter(_.nonEmpty).flatMap(_.split(","))
```
Comment (richbirch) [inline src/main/scala/uk/gov/homeoffice/drt/api/ArrivalRoutes.scala:28]: Should we be checking the port list against the ports the user has access to to guard against any inappropriate requests?

Comment (richbirch) [reply src/main/scala/uk/gov/homeoffice/drt/api/ArrivalRoutes.scala:28]: Ah ok I think I misunderstood. This is to filter down the destination ports right? So add more ports won't result in seeing arrivals that they shouldn't see

**Review by richbirch — APPROVED** [review]
Looks good to me. Nice work :champagne:

File: src/main/scala/uk/gov/homeoffice/drt/api/ArrivalRoutes.scala (hunk 1 above)
Comment (ngu04) [reply src/main/scala/uk/gov/homeoffice/drt/api/ArrivalRoutes.scala:28]: Yeah , It is jsut destination port filter . they won't see more arrivals .

**Pull request merged**